game-core/src/main/java/games/strategy/triplea/ui/MacOsIntegration.java

```diff
@@ -0,0 +1,129 @@
+ * </p>
+ */
+@Log
+public final class MacOsIntegration {
+  private MacOsIntegration() {}
+
+  /**
+   * Adds the specified about handler to the application.
+   */
+  public static void addAboutHandler(final Runnable handler) {
+    checkNotNull(handler);
+
+    addHandler(
+        getHandlerClassName("java.awt.desktop.AboutHandler", "com.apple.eawt.AboutHandler"),
+        "handleAbout",
+        "setAboutHandler",
+        args -> handler.run());
+  }
+
+  private static String getHandlerClassName(final String java9OrLaterClassName, final String java8ClassName) {
+    return isJavaVersionAtLeast9() ? java9OrLaterClassName : java8ClassName;
+  }
+
+  private static boolean isJavaVersionAtLeast9() {
+    return isJavaVersionAtLeast9(SystemProperties.getJavaSpecificationVersion());
+  }
+
+  @VisibleForTesting
+  static boolean isJavaVersionAtLeast9(final String encodedJavaSpecificationVersion) {
+    try {
+      return Float.parseFloat(encodedJavaSpecificationVersion) >= 9.0F;
+    } catch (final NumberFormatException e) {
+      log.log(Level.SEVERE, "Malformed Java specification version: '" + encodedJavaSpecificationVersion + "'", e);
+      return false;
+    }
+  }
+
+  private static void addHandler(
+      final String handlerClassName,
+      final String handlerMethodName,
+      final String addHandlerMethodName,
+      final ThrowingConsumer<Object[], Exception> handler) {
+    try {
+      addHandler(newApplication(), handlerClassName, handlerMethodName, addHandlerMethodName, handler);
+    } catch (final ReflectiveOperationException e) {
+      log.log(Level.SEVERE, "Failed to add macOS application handler: " + handlerClassName, e);
+    }
+  }
+
+  @VisibleForTesting
+  static void addHandler(
+      final Object application,
+      final String handlerClassName,
+      final String handlerMethodName,
+      final String addHandlerMethodName,
+      final ThrowingConsumer<Object[], Exception> handler) throws ReflectiveOperationException {
+    final Class<?> handlerClass = Class.forName(handlerClassName);
+    final Object handlerProxy = Proxy.newProxyInstance(
+        MacOsIntegration.class.getClassLoader(),
+        new Class<?>[] {handlerClass},
+        (proxy, method, args) -> {
+          if (handlerMethodName.equals(method.getName())) {
+            handler.accept(args);
+          }
+          return null;
+        });
+    application.getClass()
+        .getDeclaredMethod(addHandlerMethodName, handlerClass)
+        .invoke(application, handlerProxy);
+  }
+
+  private static Object newApplication() throws ReflectiveOperationException {
+    return Class.forName("com.apple.eawt.Application")
+        .getConstructor((Class[]) null)
+        .newInstance((Object[]) null);
+  }
+
+  /**
+   * Adds the specified open URI handler to the application.
+   */
+  public static void addOpenUriHandler(final Consumer<URI> handler) {
```

Comment:
@RoiEXLab Does the Mac installer do anything special to set up support for opening a `triplea:` URI like it does for Windows and Linux?  I don't see anything in _build.install4j_ but just wanted to confirm.  If possible, I'd like to be able to test all three macOS integration features with the portable build.

Reply:
The installer doesn't setup anything for Mac OS, because uri handling works different there:
While on Linux and Windows a new process is launched with the provided CLI parameter, on Mac a normal process is launched if no process is active already, and a signal is sent containing the provided parameter.

Reply:
This signal is then being passed to the handler being passed here.